Begin system spec harness

solebared opened the pull request:
Initial attempt at wiring up the frontend app to the rails backend used in system specs.

Supports two modes:
1. Automatically start `ember serve` and point it at the rails system spec backend, setting both ports from environment variables. This requires less human orchestration but is a slower dev cycle since starting the frontend app takes ~10s on each system spec run.
2. Provides a new `spec:system:start_frontend` task that spawns the frontend app with the correct port and proxy settings; system specs detect that this is already running and use it.

*Development notes*
* In order to run system specs, two new environment variables are needed. Copy over to `.env` from `backend/env-example`.
* System specs are excluded by default (in `~/.rspec`). A thin `bin/system_spec` wrapper is added to help run system specs.

*Credits*
H/T @johncarlocerna 🙌🏾 !

solebared committed:
Begin system spec harness

solebared committed:
Wire up frontend to backend for system specs

johncarlocerna commented on backend/spec/system/signup_spec.rb:
```diff
@@ -0,0 +1,8 @@
+require "system_spec_helper"
+
+RSpec.describe "signup flow" do
+  it "works" do
```
Change it block to "loads root page" or something similar

solebared committed:
Linting fixes

solebared committed:
Pass port env vars through to frontend server in dev

This supports running system specs on different frontend and backend
instances than the default ones.

solebared committed:
Drop --proxy option to `ember serve`

Appears to not be used or honored. The values in config/environment.js
seem to take effect instead.

solebared committed:
Exclude system specs by default

solebared committed:
Slight tweak to config
